From 97312cc40320192287b51dea1db2541567360d5a Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Wed, 20 Mar 2019 21:19:29 +0100
Subject: [PATCH 10/11] common: recovery: allow reading and setting time

 avc:  denied  { search } for  pid=569 comm="recovery"
 name="time" dev="sda13" ino=1048577 scontext=u:r:recovery:s0
 tcontext=u:object_r:time_data_file:s0 tclass=dir permissive=0

 avc:  denied  { read } for  pid=570 comm="recovery" name="ats_1"
 dev="sda13" ino=1048579 scontext=u:r:recovery:s0
 tcontext=u:object_r:time_data_file:s0 tclass=file permissive=0

 avc:  denied  { sys_time } for  pid=567 comm="recovery"
 capability=25 scontext=u:r:recovery:s0 tcontext=u:r:recovery:s0
 tclass=capability permissive=0

Change-Id: I3a5b2f5cc749fbee56c17765042b9dc616d4fedd
---
 common/private/recovery.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/common/private/recovery.te b/common/private/recovery.te
index 97d3c1e..26b1df9 100644
--- a/common/private/recovery.te
+++ b/common/private/recovery.te
@@ -31,4 +31,8 @@ domain_trans(recovery, otapreopt_chroot_exec, backuptool)
 
 # Fix reboot?
 allow recovery self:netlink_kobject_uevent_socket create_socket_perms_no_ioctl;
+
+# Reading and setting time
+r_dir_file(recovery, time_data_file)
+allow recovery self:capability sys_time;
 ')
-- 
2.17.1

